docs/rfc/rfc-005-authentication.md

```diff
@@ -0,0 +1,132 @@
+# RFC 005: Authentication
+
+## Changelog
+
+- 09/05/2023: DRAFT
+
+## Problem Statement
+
+The recently proposed changes to the account system in the Cosmos SDK (as described in the Accounts RFC) focus on
+transforming accounts into more sophisticated, module-like entities capable of managing their own state. However, several
+key issues need to be addressed with respect to authentication:
+
+### 1. Account Abstraction
+
+The Accounts RFC provides an abstraction over accounts as business logic entities, similar to modules, but does not
+address how authentication should be handled for these new account types.
+
+### 2. Accounts without Credentials
+
+It is possible for accounts to exist without any associated credentials, which means they cannot initiate state 
+transitions through transactions. A mechanism is needed to attach authentication credentials to such accounts to enable
+them to interact with the network.
+
+### 3. Pluggable Authentication Credentials
+
+The authentication system should provide a way to plug different credential mechanisms into accounts. This would allow 
+developers to customize the authentication process for different account types or use cases.
+
+### 4. Extensible Credential Mechanisms
+   
+Currently, the Cosmos SDK primarily supports accounts backed by public/private key pairings. However, there is a growing
+need to design and implement an extensible system that can support abstracted credentials, which are not limited to
+public/private key pairings. This would enable the development of more advanced and flexible authentication systems that
+cater to a broader range of use cases and security requirements.
+
+To address these problems, the Cosmos SDK needs to develop an authentication system that is closely integrated with the
+proposed account system, providing a robust and extensible framework for managing authentication credentials and supporting
+a variety of credential mechanisms.
+
+## Implementation
+
+We have initially mentioned that `x/accounts` only provides for a way to deploy accounts, but they're not backed by any
+credential, which means no state transition on behalf of the account can be incepted from a TX.
+
+`x/authn` is meant to provide the link between accounts and their TX credentials.
+Specifically `x/authn` defines the following `tx` gRPC interface:
+
+```protobuf
+// Msg defines the Msg service.
+service Msg {
+  rpc CreateAuthenticatedAccount(MsgCreateAuthenticatedAccount) returns (MsgCreateAuthenticatedAccountResponse) {};
+  rpc UpdateCredentials(MsgUpdateCredentials) returns (MsgUpdateCredentialsResponse) {};
+  rpc DeleteCredentials(MsgDeleteCredentials) returns (MsgDeleteCredentials) {};
+}
+message MsgCreateAuthenticatedAccount {
+  string sender = 1;
+  google.Protobuf.Any credential = 2;
+  accounts.MsgDeploy deploy_msg = 3;
+}
+message MsgCreateAuthenticatedAccountResponse {
+  accounts.MsgDeployResponse deploy_response = 1;
+}
+message MsgUpdateCredentials {
```

Comment:
This seems to imply there is only one instance of `Credential` per account. This is fine as long as credentials can define complex behaviour, but I think at least two "credentials aggregators" should be available by default: `AnyOf`, `AllOf`.